crates/bls-crypto/src/hash_to_curve/try_and_increment.rs

```diff
@@ -100,6 +99,22 @@ where
             // produce a hash with sufficient length
             let candidate_hash = self.hasher.hash(domain, msg, hash_bytes)?;
 
+            // handle the Celo deployed bit extraction logic
+            cfg_if::cfg_if! {
+                if #[cfg(feature = "compat")] {
+                    use algebra::serialize::{Flags, SWFlags};
+
+                    let mut candidate_hash = candidate_hash[..num_bytes].to_vec();
+                    let positive_flag = candidate_hash[num_bytes - 1] & 2 != 0;
+                    if positive_flag {
+                        candidate_hash[num_bytes - 1] |= SWFlags::PositiveY.u8_bitmask();
+                    } else {
+                        candidate_hash[num_bytes - 1] &= !SWFlags::PositiveY.u8_bitmask();
+                    }
+                    let candidate_hash = candidate_hash;
```

Comment:
Is this assignment needed?

Reply:
Just to remove the mut to have equivalent code :)